### ISSUE TITLE
number of class and size issue

### ISSUE THREAD

**Issue by asifsetu** [kdijkstra13/OpenCentroidNet#1]
I have changed the create dataset function with the following:
def create_dataset(folder, file: str = "training.csv", n: int = 50, height: int = 384, width: int = 384, n_circles: int = 5, min_r: int = 15, max_r: int = 20, prefix="train", n_classes=1):

I also changed in the config file,
num_classes = 2

While run the train.py I got the follwoing error. Could you please help me to solve the issue.

![image](https://user-images.githubusercontent.com/28484034/70807304-955ac600-1dbd-11ea-80bb-10c3d970a6ce.png)

**Comment by klaasdijkstra** [comment]
I cannot reproduce your error when training/validating 1 class (when making the changes you propose). Does it work when you use the original code?

Python: 3.7.5
PyTorch: 1.3.1
NumPy: 1.17.3

This is my output:

`Training 50 images for 500 epochs with a batch size of 20.
Validate 10 images each 10 epochs and learning rate 0.001.

0%|                                                   | 0/499 [00:00<?,?it/s]/home/klaas/anaconda3/envs/py37/lib/python3.7/site-packages/torch/nn/functional.py:2404: UserWarning: nn.functional.upsample is deprecated. Use nn.functional.interpolate instead.
Epoch 10/500 Loss(T): 0.161153 and Loss(V): 9999.00000:   2%| | 9/499 [00:05<04:29,  1.82it/s]Update model with loss 0.09310887008905411
Epoch 20/500 Loss(T): 0.046168 and Loss(V): 0.09311:   4%| | 19/499 [00:11<04:33,  1.75it/s]Update model with loss 0.04768607020378113
Epoch 30/500 Loss(T): 0.032205 and Loss(V): 0.04769:   6%| | 29/499 [00:17<04:28,  1.75it/s]Update model with loss 0.036768149584531784
Epoch 50/500 Loss(T): 0.031167 and Loss(V): 0.04132:  10%| | 49/499 [00:28<04:08,  1.81it/s]Update model with loss 0.03646500036120415
Epoch 60/500 Loss(T): 0.014757 and Loss(V): 0.03647:  12%| | 59/499 [00:34<03:57,  1.85it/s]Update model with loss 0.01476603839546442
Epoch 70/500 Loss(T): 0.010603 and Loss(V): 0.01477:  14%|▏| 69/499 [00:40<04:01,  1.78it/s]Update model with loss 0.009750930592417717
Epoch 80/500 Loss(T): 0.009481 and Loss(V): 0.00975:  16%|▏| 79/499 [00:46<03:45,  1.86it/s]Update model with loss 0.007566744461655617
Epoch 110/500 Loss(T): 0.008259 and Loss(V): 0.00824:  22%|▏| 109/499 [01:04<03:32,  1.84it/s]Update model with loss 0.006477628834545612
Epoch 180/500 Loss(T): 0.006625 and Loss(V): 0.00796:  36%|▎| 179/499 [01:44<03:01,  1.77it/s]Update model with loss 0.006435800809413195
Epoch 230/500 Loss(T): 0.006794 and Loss(V): 0.00718:  46%|▍| 229/499 [02:14<02:32,  1.77it/s]Update model with loss 0.005504772998392582
Epoch 300/500 Loss(T): 0.005792 and Loss(V): 0.00648:  60%|▌| 299/499 [02:54<01:48,  1.85it/s]Update model with loss 0.005496438127011061
Epoch 320/500 Loss(T): 0.005521 and Loss(V): 0.00625:  64%|▋| 319/499 [03:06<01:39,  1.81it/s]Update model with loss 0.005316810682415962
Epoch 440/500 Loss(T): 0.006253 and Loss(V): 0.00753:  88%|▉| 439/499 [04:16<00:33,  1.81it/s]Update model with loss 0.004992243833839893
Epoch 450/500 Loss(T): 0.007215 and Loss(V): 0.00499:  90%|▉| 449/499 [04:22<00:27,  1.80it/s]Update model with loss 0.004575344268232584
Epoch 499/500 Loss(T): 0.006552 and Loss(V): 0.00787: 100%|█| 499/499 [04:51<00:00,  1.71it/s]
Save snapshot to: /home/klaas/Projects/pc_deploy/OpenCentroidNetGitHub/data/CentroidNet.pth
Load snapshot from: /home/klaas/Projects/pc_deploy/OpenCentroidNetGitHub/data/CentroidNet.pth
Predicting 10 files with loss <class 'centroidnet.centroidnet_core.CentroidLoss'>
  0%|                                                     | 0/2 [00:00<?, ?it/s]/home/klaas/Projects/pc_deploy/OpenCentroidNetGitHub/train.py:217: FutureWarning: arrays to stack must be passed as a "sequence" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.
  result_images, result_centroids = predict(validation_set, model, loss, max_dist=Config.max_dist, binning=Config.binning, nm_size=Config.nm_size, centroid_threshold=Config.centroid_threshold)
100%|█████████████████████████████████████████████| 2/2 [00:00<00:00,  `4.75it/s]`

**Comment by asifsetu** [comment]
Many thanks for your help. It works fine when I create dataset using create_dataset.py. However, it provides the above error while I used my own dataset with 384X384 dimension 1 channel and 1 class object. 
Anyway, could you please suggest any better tools that I can create csv file for training? Thanks.

**Comment by klaasdijkstra** [comment]
This is probably specific to your data set. My guess is that the one channel images are handled incorrectly. A possible workaround is to use a multi channel image. Could you try to create a two or three channel image? In your case you could probably just copy the one channel into two channels.
